exchanges/account/account_types.go

```diff
@@ -34,7 +34,11 @@ type Accounts struct {
 	// TODO: Credential tracker to match to keys that are managed and return
 	// pointer.
 	// TODO: Have different cred struct for centralized verse DEFI exchanges.
-	SubAccounts map[Credentials]map[key.SubAccountCurrencyAsset]*ProtectedBalance
+	subAccounts map[Credentials]map[key.SubAccountAsset]subAccountAsset
+}
+
+type subAccountAsset struct {
+	balances map[*currency.Item]*ProtectedBalance
 }
```

Comment:
🚧  
* This structure actually has nothing to do with subAccounts anymore. And doesn't actually imply anything about assets either. So `currencyBalances` is a better name
* Might be missing something, but why do we need it to be a struct 🤷 
```suggestion
type currencyBalances = map[*currency.Item]*ProtectedBalance
```

Reply:
In the current context, currencyBalances seems like a good name. I'm considering introducing account position data in the future, so using a struct allows for better extensibility.

Reply:
Got it! We'll keep it as a map type for now. We can revisit and change it to a struct later when handling the position update.

Reply:
Happy with this 👍 Thanks